API/Extensions/FileUploader.cs

```diff
@@ -0,0 +1,150 @@
+using AngleSharp;
+using AngleSharp.Io;
+using API.Resources;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using File = Models.File;
+
+namespace API.Extensions
+{
+    /// <summary>
+    /// File already exists exception
+    /// </summary>
+    [Serializable]
+    public class FileExistException : Exception
+    {
+        /// <summary>
+        /// File already exist constructor
+        /// </summary>
+        /// <param name="name"></param>
+        public FileExistException(string name)
+            : base(String.Format("File {0} already exists", name))
+        { }
+    }
+
+    /// <summary>
+    /// Interface for file uploader
+    /// </summary>
+    public interface IFileUploader
+    {
+
+        /// <summary>
+        /// Remove specials characters from string
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns> String without special characters</returns>
+        string RemoveSpecialCharacters(string str);
+
+
+        /// <summary>
+        /// Uploads single file
+        /// </summary>
+        /// <param name="file"></param>
+        /// <param name="fileName"></param>
+        /// <returns> path of file location </returns>
+        Task<string> UploadSingleFile(IFormFile file, string fileName);
+
+        /// <summary>
+        /// Method deletes the file from the file server
+        /// </summary>
+        /// <param name="file"></param>
+        void DeleteFile(File file);
+
+    }
+
+    /// <summary>
+    /// Class which is responsible for uploading files
+    /// </summary>
+    public class FileUploader : IFileUploader
+    {
+
+        private readonly string uploadPath;
+        private readonly string serverRoot;
+
+        /// <summary>
+        /// File Uploader
+        /// </summary>
+        /// <param name="env"></param>
+        public FileUploader(IWebHostEnvironment env)
+        {
+            uploadPath = Path.Combine(env.WebRootPath, "Resources\\");
+            serverRoot = Path.Combine("https://localhost:5001/Resources/");
```

Comment:
```suggestion
            serverRoot = Path.Combine("https://localhost:5001/Resources/");
```
Never use hardcoded values like these. This should probably be a setting in appsettings (both dev and prod)

Reply:
removed lines as frontend is responsible for this.